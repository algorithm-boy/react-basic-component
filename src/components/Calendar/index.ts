import Calendar from './Calendar';

export { CalendarProps, CalendarChange } from './Calendar';
export default Calendar;
