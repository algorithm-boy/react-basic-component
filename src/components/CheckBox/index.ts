import CheckBox from './CheckBox';
export { CheckBoxProps } from './CheckBox';
export default CheckBox;
