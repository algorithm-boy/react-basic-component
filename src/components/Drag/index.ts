import Drag from './Drag';

export { DragProps } from './Drag';
export default Drag;
