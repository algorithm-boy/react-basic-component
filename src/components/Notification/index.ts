import Notification from './Notification';

export { NotificationProps } from './Notification';
export default Notification;
