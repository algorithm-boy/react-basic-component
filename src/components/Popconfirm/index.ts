import Popconfirm from './Popconfirm';

export { PopconfirmProps } from './Popconfirm';
export default Popconfirm;
