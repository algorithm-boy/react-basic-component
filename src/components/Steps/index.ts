import Steps from './Steps';
import Step from './Step';

Steps.Step = Step;

export { Step };
export { StepProps } from './Step';
export { StepsProps } from './Steps';
export default Steps;
