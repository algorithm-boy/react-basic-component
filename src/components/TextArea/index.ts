import TextArea from './TextArea';

export { TextAreaProps, TextAreaStates } from './TextArea';
export default TextArea;
