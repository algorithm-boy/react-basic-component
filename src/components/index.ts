import Upload from './Upload';
import Calendar from './Calendar';
import Select from './Select';
import Notification from './Notification';
import Switch from './Switch';
import Button from './Button';
import Steps from './Steps';
import Portal from './Portal';
import Drag from './Drag';
import CheckBox from './CheckBox';
import Progress from './Progress';
import Tooltip from './Tooltip';
import Popconfirm from './Popconfirm';
import TextArea from './TextArea';
import Input from './Input';

export default { Upload, Calendar, Select, Switch, Notification, Button, Steps, Portal, Drag, CheckBox, Popconfirm, Tooltip, Progress, TextArea, Input };

export { default as Upload, UploadProps } from './Upload';
export { default as Calendar, CalendarProps, CalendarChange } from './Calendar';
export { default as Select, SelectProps, Option, IOptProps } from './Select';
export { default as Switch, SwitchProps } from './Switch';
export { default as Notification, NotificationProps } from './Notification';
export { default as Button, ButtonProps } from './Button';
export { default as Steps, Step, StepProps, StepsProps } from './Steps';
export { default as CheckBox, CheckBoxProps } from './CheckBox';
export { default as Portal, PortalProps } from './Portal';
export { default as Drag, DragProps } from './Drag';
export { FormField, Form, FormStore, useFormStore, useFormChange } from './FieldForm';
export { default as Progress } from './Progress';
export { ProgressProps } from './Progress';
export { default as Popconfirm, PopconfirmProps } from './Popconfirm';
export { default as Tooltip, TooltipProps } from './Tooltip';
export { default as TextArea, TextAreaProps } from './TextArea';
export { default as Input, InputProps } from './Input';
