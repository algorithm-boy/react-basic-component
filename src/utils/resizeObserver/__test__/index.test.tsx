import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactResizeObserver from '../';
import { TextArea } from '../../../components';

configure({ adapter: new Adapter() });

describe('ReactResizeObserver - renders', function () {
  it('renders correctly - render', () => {
    const wrapper = render(
      <ReactResizeObserver>
        <TextArea
          placeholder='placeholder'
          defaultValue='defaultValue'
        />
      </ReactResizeObserver>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly - mount', () => {
    const wrapper = mount(
      <ReactResizeObserver>
        <TextArea
          placeholder='placeholder'
          value='value'
        />
      </ReactResizeObserver>
    );
    expect(wrapper.contains(<TextArea
      placeholder='placeholder'
      value='value'
    />)).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ReactResizeObserver - parameters', () => {
  it('parameters check', () => {
    const wrapper = mount(<ReactResizeObserver
      disabled
    >
      <TextArea
        placeholder='placeholder'
        value='value'
      />
    </ReactResizeObserver>);

    expect(wrapper.props().disabled).toEqual(true);
  });
});
