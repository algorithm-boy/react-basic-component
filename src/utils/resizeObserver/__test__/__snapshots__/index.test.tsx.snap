// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactResizeObserver - renders renders correctly - mount 1`] = `ReactWrapper {}`;

exports[`ReactResizeObserver - renders renders correctly - render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "hfdc-textarea",
      "maxlength": "Infinity",
      "placeholder": "placeholder",
      "rows": "1",
      "style": "resize:none",
    },
    "children": Array [
      Object {
        "data": "defaultValue",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "textarea",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "maxlength": undefined,
      "placeholder": undefined,
      "rows": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "maxlength": undefined,
      "placeholder": undefined,
      "rows": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
